src/nvim/tui/tui.c

```diff
@@ -384,8 +386,13 @@ static void terminfo_stop(TUIData *tui)
   unibi_out_ext(tui, tui->unibi_ext.disable_extended_keys);
   // May restore old title before exiting alternate screen.
   tui_set_title(tui, (String)STRING_INIT);
-  // Exit alternate screen.
-  unibi_out(tui, unibi_exit_ca_mode);
+  // if nvim exited with nonzero status, without indicated this was an
+  // intentional exit (like `:1cquit`), it likely was an internal failure.
+  // Don't clobber the stderr error message in this case.
+  if (ui_client_exit_status == tui->seen_error_exit) {
```

Comment:
Can we use [v:exiting](https://github.com/neovim/neovim/blob/a293e2794dcf65a35eda6eeea6ea480bfef9dc73/src/nvim/main.c#L697) instead of adding `ui_client_exit_status`? Could init `seen_error_exit = INT_MAX`, then condition could be `if (tui->seen_error_exit != INT_MAX)`